bucket/siyuan-note.json

```diff
@@ -0,0 +1,47 @@
+{
+    "version": "1.4.8",
+    "description": "SiYuan is a Markdown Block-Reference and Bidirectional-Link note-taking application",
+    "homepage": "https://b3log.org/siyuan",
+    "license": {
+        "identifier": "Proprietary",
+        "url": "https://b3log.org/siyuan/eula.html"
+    },
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/siyuan-note/siyuan/releases/download/v1.4.8/siyuan-1.4.8-win.exe#/siyuan.7z",
+            "hash": "2e72af2c6211595ce4e47c2141e029ce53fb7f934d383f9d8f79493ac6a70b61",
+            "pre_install": [
+                "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-64.7z\" \"$dir\"",

```

Comment:
Could you explain why you need pre-install?

Why would extract_dir not work for example?

Reply:
Here `pre-install` works the same as `installer.script`.
Shall I replace it with the code below?
```json
"64bit": {
    "url": "https://github.com/siyuan-note/siyuan/releases/download/v1.4.8/siyuan-1.4.8-win.exe#/dl.7z",
    "hash": "2e72af2c6211595ce4e47c2141e029ce53fb7f934d383f9d8f79493ac6a70b61",
    "installer": {
        "script": [
            "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-64.7z\" \"$dir\"",
            "Remove-Item \"$dir\\`$*\" -Recurse"
        ]
    }
},
"32bit": {
    "url": "https://github.com/siyuan-note/siyuan/releases/download/v1.4.8/siyuan-1.4.8-win32.exe#/dl.7z",
    "hash": "7f97024325040287688ab0ea12ffd8b0e0045680e82c734461254e6e382fd53a",
    "installer": {
        "script": [
            "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-32.7z\" \"$dir\"",
            "Remove-Item \"$dir\\`$*\" -Recurse"
        ]
    }
}
```

Reply:
No I would like to know why we need pre_installer?

Why are the existing approaches not working?

Like a lot of packages have this and am a bit curious

Reply:
Actually, I'm confused, too.
I'm just a scoop novick who wants to contribute to a favorite project.😂
These configurations are moved from another niche bucket.👀
[dorado/siyuan.json at master · chawyehsu/dorado](https://github.com/chawyehsu/dorado/blob/master/bucket/siyuan.json)